Migrate exportdb to torch.export from torchdynamo.export

tugsbayasgalan opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #103861
* #102310

Things that needed to be fixed:
1. Fix a bug with returning dict output type 
2. Make pass_base work with map implementation
3. Fix subtle bug with dynamo not propagating "val" in node.meta
4. Add export_constraints field in ExportCase in ExportDB

cc @voznesenskym @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @ipiszy @chenyang78

tugsbayasgalan committed:
[WIP] Migrate exportdb to torch.export from torchdynamo.export

[ghstack-poisoned]

tugsbayasgalan committed:
Update on "[WIP] Migrate exportdb to torch.export from torchdynamo.export"




[ghstack-poisoned]

tugsbayasgalan committed:
Update on "[WIP] Migrate exportdb to torch.export from torchdynamo.export"




[ghstack-poisoned]

tugsbayasgalan committed:
Update on "[WIP] Migrate exportdb to torch.export from torchdynamo.export"




cc voznesenskym penguinwu anijain2305 EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx ipiszy chenyang78

[ghstack-poisoned]

tugsbayasgalan commented on torch/_export/pass_base.py:
```diff
@@ -326,16 +326,16 @@ def call_cond(
-            control_flow.map,
-            (f_branch.graph_module, xs, *args),
+            torch.ops.map_impl,
+            (f_branch.graph_module, num_args, *args),
```
@ydwu4 Is this correct?

tugsbayasgalan replied:
ok sounds good

zhxchen17 commented on torch/_export/db/case.py:
```diff
@@ -77,6 +78,7 @@ class ExportCase:
     # Tags associated with the use case. (e.g dynamic-shape, escape-hatch)
     tags: Set[str] = field(default_factory=lambda: set())
     support_level: SupportLevel = SupportLevel.SUPPORTED
+    export_constraints: List[Constraint] = field(default_factory=lambda: [])
```
```suggestion
    constraints: List[Constraint] = field(default_factory=lambda: [])
```

zhxchen17 reviewed (approved):
thanks!

tugsbayasgalan committed:
Update on "Migrate exportdb to torch.export from torchdynamo.export"


Things that needed to be fixed:
1. Fix a bug with returning dict output type 
2. Make pass_base work with map implementation
3. Fix subtle bug with dynamo not propagating "val" in node.meta
4. Add export_constraints field in ExportCase in ExportDB

cc voznesenskym penguinwu anijain2305 EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx ipiszy chenyang78

[ghstack-poisoned]

tugsbayasgalan commented:
@pytorchbot merge -c

tugsbayasgalan commented:
@pytorchbot merge -ic

tugsbayasgalan commented:
@pytorchbot merge -i

tugsbayasgalan commented:
@pytorchbot label "topic: not user facing"

tugsbayasgalan commented:
@pytorchbot merge -i

huydhn commented:
@pytorchbot revert -m 'It looks like this change is failing in trunk due to a landrace https://hud.pytorch.org/pytorch/pytorch/commit/fb6173a4ac60ed5a22cff2c68134633eb72e53b9' -c landrace
